rust/src/xenctrl.rs

```diff
@@ -169,4 +173,39 @@ impl Xenctrl {
       }
     }
   }
+
+  pub fn list_domains (&self) -> Result<Vec<DomainInfo>> {
+    unsafe {
+      let max_doms: u32 = 1024;
+      let mut info: Vec<xenctrl_sys::xc_domaininfo_t> = Vec::with_capacity(max_doms as usize);
+      info.resize_with(max_doms as usize, Default::default);
+      let mut dom_id = 0;
+      let mut res: Vec<xenctrl_sys::xc_domaininfo_t> = Vec::new();
```

Comment:
Maybe rename `res` to `domains` to be more clear?

Reply:
Can you try without the type specifier? I suppose the inference algorithm can be smart here.

Reply:
indeed the inference works here! :)